src/control/server/engine/exec.go

```diff
@@ -128,12 +136,16 @@ func (r *Runner) Signal(signal os.Signal) error {
 
 	r.log.Debugf("Signalling I/O Engine instance %d (%s)", r.Config.Index, signal)
 
+	r.cmdMut.Lock()
```

Comment:
Same question as above.

Reply:
this one probably could be because the Signal() call does not write an internal state.